### PR TITLE
Payday for monthly emp

### PR DESCRIPTION
**Pull request by octopusinvitro**
The paying calculation is quite long so I decided to break it into little steps. Since there are a lot of new objects that I had to add, I started trying to pay a monthly employee first.

For that I had to create:
- A `calculatePay()` method in the payment type interface. At this point it's only implemented for the monthly payment, in the sense that the other methods just return zero for the moment.
- An `isPayDay()` method in the payment schedule interface. Again, it's only implemented for the monthly schedule, the others return a hardcoded false.
- The payment method interface is still empty, but it will contain a `pay()` method that I didn't have to implement yet.

Also, introducing the possibility to pay employees has made me have to take care of the dates at this point, since they are needed for the `isPayDay()` check. Thanks god that @christophgockel [suggested to use `LocalDate`](https://github.com/me-stevens/payroll/pull/3#discussion_r49067127), which made things pretty easy (I was going to use `Calendar`):

``` java
    private boolean isLastDayOfMonth(LocalDate date) {
        Month month     = date.getMonth();
        Month nextMonth = date.plusDays(1).getMonth();
        return month != nextMonth;
    }
```

I had to add a `Paycheck` class too. Also, the deductions due to affiliations of the employee are not taken into account yet in the `getPaid()` method of the `Employee` class, so I am just returning the net pay. This will change in the future and will affect the `Paycheck` class too.

Finally, I also added methods in the employee class to calculate its pay (`getPaid()`) and to check if it is payday (`isPayDay()`), which are used in the `PayDayTransaction`.

@ecomba

### CODE REVIEW

File: src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Test
+    public void returnsFalseIfNotLastDayOfMonth() {
+        LocalDate notLastDayOfMonth = LocalDate.of(2016, JANUARY, 30);
+        assertFalse(monthlySchedule.isPayDay(notLastDayOfMonth));
+    }
+
+    @Test (expected = DateTimeException.class)
+    public void throwsExceptionIfInvalidDate() {
```
Comment (christophgockel) [inline src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java]: This method doesn't test your `MonthlySchedule`.
The expected exception is thrown in `LocalDate.of()`.

File: src/main/java/com/mael/payroll/transactions/PayDayTransaction.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private Map<Integer, Paycheck> payChecks;
+
+    public PayDayTransaction(PayrollDBFacade payrollDB, LocalDate payDay) {
+        this.payrollDB = payrollDB;
+        this.payDay    = payDay;
+        this.payChecks = new HashMap<>();
+    }
+
+    public void execute() {
+        for (Map.Entry<Integer, Employee> entry : payrollDB.getAllEmployees().entrySet()) {
```
Comment (christophgockel) [inline src/main/java/com/mael/payroll/transactions/PayDayTransaction.java]: `entry` provides too little information about the actual object it represents.

File: src/test/java/com/mael/payroll/transactions/PayDayTransactionTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    public void paysMonthlyEmployeeAtTheEndOfTheMonth() {
+        AddEmployee addMonthlyEmployee = new AddMonthlyEmployee(payrollDB, 1, "Squiddo", "FishBowl", 1000.0);
+        addMonthlyEmployee.execute();
+
+        LocalDate payDay = of(2016, JANUARY, 31);
+        PayDayTransaction payDayTransaction = new PayDayTransaction(payrollDB, payDay);
+        payDayTransaction.execute();
+
+        Paycheck paycheck = payDayTransaction.getPayChecks().get(1);
```
Comment (christophgockel) [inline src/test/java/com/mael/payroll/transactions/PayDayTransactionTest.java]: I the employee ID is used here to get a paycheck from a certain employee I would give it a proper name.
At first sight it looks like you're getting the second element from a list there.

File: src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java (hunk 1 above)
Comment (octopusinvitro) [inline src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java]: I'm sure something was happening in my head to think that was a good idea XD
Corrections were added :+1:

File: src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    private MonthlySchedule monthlySchedule;
+
+    @Before
+    public void setUp() {
+        monthlySchedule = new MonthlySchedule();
+    }
+
+    @Test
+    public void returnsTrueIfLastDayOfMonth() {
```
Comment (christophgockel) [inline src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java:22]: The `returnsTrue...` and `returnsFalse...` reads a bit too technical for me.
Is there a way to express the behaviour you're verifying in a way that is it more descriptive?
In the end test names act as documentation for us and our team.

Comment (octopusinvitro) [inline src/test/java/com/mael/payroll/paymentSchedules/MonthlyScheduleTest.java:22]: :+1: Good catch
Changed them to `itIsPayDayIfLastDayOfMonth()` and `isNotPayDayIfNotLastDayOfMonth()`

**Pull request closed without merging**